### PR TITLE
[SPARK-34728][SQL] Remove all SQLConf.get if extends from SQLConfHelper

### PR DESCRIPTION
**Pull request by leoluan2009**
### What changes were proposed in this pull request?

Remove all SQLConf.get to conf if extends from SQLConfHelper

### Why are the changes needed?

Clean up code.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing unit tests.

### CODE REVIEW

**Comment by leoluan2009** [comment]
cc @wangyum

**Comment by wangyum** [comment]
ok to test.

**Comment by wangyum** [comment]
There are still some rules that have not been cleaned up:
https://github.com/apache/spark/blob/9e6882feca0800d5d4f9920886cb5dae73bbe1d4/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CTESubstitution.scala#L34
https://github.com/apache/spark/blob/e3058ba17cb4512537953eb4ded884e24ee93ba2/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DecimalPrecision.scala#L84
https://github.com/apache/spark/blob/66c5e0132209a5a94f9d7efb5e895f143b0ef53b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveUnion.scala#L101
https://github.com/apache/spark/blob/5d9cfd727c21f0f50f92c0236cdeb20f3b9111dc/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala#L275
https://github.com/apache/spark/blob/32a523b56fc329e858f3a463314c13e3710be5e0/sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala#L225

File: sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala [hunk 1]
```diff
@@ -97,7 +96,7 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper {
           filteringPlan,
           joinKeys,
           index,
-          SQLConf.get.dynamicPartitionPruningReuseBroadcastOnly || !hasBenefit),
+          ç.dynamicPartitionPruningReuseBroadcastOnly || !hasBenefit),
```
Comment (wangyum) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala:99]: typo?

**Comment by wangyum** [comment]
cc @cloud-fan

**Review by HyukjinKwon — COMMENTED** [review]
Looks good

**Comment by cloud-fan** [comment]
looks good, let's fix all the places.

**Comment by maropu** [comment]
retest this please

**Comment by maropu** [comment]
Looks fine, too.

**Comment by wangyum** [comment]
Rename `Remove all SQLConf.get if extends from Rule` to `Remove all SQLConf.get if extends from SQLConfHelper`?

**Review by wangyum — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Review by imback82 — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**